src/test/java/org/sonarsource/sonarlint/ls/mediumtests/JavaMediumTests.java

```diff
@@ -107,7 +109,40 @@ public void analyzeSimpleJavaFileReuseCachedClasspath() throws Exception {
   }
 
   @Test
-  public void analyzeSimpleJavaTestFileOnOpen() throws Exception {
+  void analyzeSimpleJavaFilePassVmClasspath() throws Exception {
+    Path javaHome = Paths.get(System.getProperty("java.home"));
+    Path currentJdkHome = javaHome.endsWith("jre") ? javaHome.getParent() : javaHome;
+    boolean isModular = Files.exists(currentJdkHome.resolve("lib/jrt-fs.jar"));
+
+    emulateConfigurationChangeOnClient("", true, true, true);
+
+    String uri = getUri("analyzeSimpleJavaFileOnOpen.java");
+
+    GetJavaConfigResponse javaConfigResponse = new GetJavaConfigResponse();
+    javaConfigResponse.setSourceLevel("1.8");
+    javaConfigResponse.setTest(false);
+    javaConfigResponse.setClasspath(new String[0]);
+    javaConfigResponse.setVmLocation(currentJdkHome.toString());
+    client.javaConfigs.put(uri, javaConfigResponse);
+
+    List<Diagnostic> diagnostics = didOpenAndWaitForDiagnostics(uri, "java", "public class Foo {\n  public static void main() {\n  // System.out.println(\"foo\");\n}\n}");
```

Comment:
Nitpick: I would love to see a test source where the difference in classpath yields in a difference in analysis results. Might be tricky to pull out.